### PR TITLE
Fix nightly CircleCI

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix a few nightly CircleCI issues

The effect could be found [here](https://app.circleci.com/pipelines/github/huggingface/transformers/50144/workflows/aaa65460-bbb5-4048-af3e-9450af06e231).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -81,11 +81,13 @@ jobs:
             - run: pip install .
             - run: |
                   mkdir test_preparation
-                  echo "tests" > test_preparation/test_list.txt
-                  echo "tests" > test_preparation/examples_test_list.txt
-            - run: python utils/tests_fetcher.py --filter_tests
-            - run: mv test_list.txt test_preparation/filtered_test_list.txt
-            - run: mv test_repo_utils.txt test_preparation/test_repo_utils.txt
+                  echo -n "tests" > test_preparation/test_list.txt
```
Comment (ydshieh) [inline .circleci/config.yml:84]: without `-n`, The command `echo` will write a `\n` to the file, which makes the condition `test_files == ["tests"]` False in `filter_tests`, and eventually failed at `f.split(os.path.sep)[1]` with index error.

File: .circleci/config.yml [hunk 2]
```diff
@@ -81,11 +81,13 @@ jobs:
...
+                  echo -n "tests" > test_preparation/examples_test_list.txt
+            - run: |
+                  echo -n "tests" > test_list.txt
```
Comment (ydshieh) [inline .circleci/config.yml:87]: We need to have this file to make the next line below to find a file to filter.

File: .circleci/config.yml [hunk 3]
```diff
@@ -81,11 +81,13 @@ jobs:
...
+                  python utils/tests_fetcher.py --filter_tests
+                  mv test_list.txt test_preparation/filtered_test_list.txt
+            - run: echo -n "tests/repo_utils" > test_preparation/test_repo_utils.txt
```
Comment (ydshieh) [inline .circleci/config.yml:90]: `filter_tests` won't reproduce `test_repo_utils.txt` if I understand correctly. So we need to provide here manually

Comment (sgugger) [reply .circleci/config.yml:90]: Let's put this one in the previous block (so line 86)!

**Review by sgugger — APPROVED** [review]
Thanks for all the fixes! Another needed from the logs I received this weekend is that we need GitPython installed, so we need to add `- run: pip install GitPython` like in the regular fetch_tests job. Can you add it in this PR?

**Comment by ydshieh** [comment]
@sgugger I accidentally push `GitPython` to `main`

https://github.com/huggingface/transformers/commit/6f8064da6b0c8f003731f292acc64f281a2aea65

😨

File: .circleci/config.yml (hunk 3 above)
Comment (ydshieh) [reply .circleci/config.yml]: done

**Comment by sgugger** [comment]
Ah so this is done, perfect!

**Pull request merged**